cpp/src/column/column_factories.cpp

```diff
@@ -128,20 +128,23 @@ struct column_from_scalar_dispatch {
              rmm::mr::device_memory_resource* mr,
              cudaStream_t stream) const
   {
-    auto null_mask = create_null_mask(size, mask_state::ALL_NULL, stream, mr);
     if (!value.is_valid())
-      return std::make_unique<column>(
-        value.type(), size, rmm::device_buffer{}, std::move(null_mask), size);
+      return std::make_unique<column>(value.type(),
+                                      size,
+                                      rmm::device_buffer{0, stream, mr},
+                                      create_null_mask(size, mask_state::ALL_NULL, stream, mr),
+                                      size);
 
     // Create a strings column_view with all nulls and no children.
     // Since we are setting every row to the scalar, the fill() never needs to access
     // any of the children in the strings column which would otherwise cause an exception.
+    auto null_mask = create_null_mask(size, mask_state::ALL_NULL, stream);
```

Comment:
Consider making the same change here as you did above? aka removing the temporary

Reply:
I don't think I can do it here, the mask buffer is used in multiple statements. The ownership is not transferred in the `column_view sc` creation, so I have to keep `null_mask` alive until `strings::detail::fill` is done.

Reply:
Mmm i see. Makes sense